### PR TITLE
Improve the AWS module to generate isolated resources on execution part 1

### PR DESCRIPTION
**Pull request by EduLeon12**
## Description 
| Related issue |
| -- |
| Closes https://github.com/wazuh/wazuh-qa/issues/4714  |

As part of https://github.com/wazuh/wazuh/issues/17788 the issue https://github.com/wazuh/wazuh-qa/issues/4714 was created to generate isolated resources on execution. 

This PR adds all the necessary functions, fixtures to interact with the qa-integration-framework for the creation of the resources. It also adds the configurator who acts as a handler that configures the test and its metadata to allow the creation of the correct resource for each test.

> [!IMPORTANT]
> Due to internal team changes this is the first part of the issue but it will be continued on another issue.

### Tests
Since the only test suites implemented were the `test_basic` and the `test_custom_buckets` are the only ones worth displaying in this section.

```python
root@vagrant:/wazuh/tests/integration/test_aws# pytest -sx --disable-warnings test_basic.py
================================ test session starts ================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.3.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.0.0, html-3.1.1
collected 16 items                                                          

test_basic.py ................

==================== 16 passed, 2 warnings in 294.42s (0:04:54) =====================

root@vagrant:/wazuh/tests/integration/test_aws# pytest -sx --disable-warnings test_custom_bucket.py
======================== test session starts =========================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.3.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.0.0, html-3.1.1
collecting ... 
collected 2 items

test_custom_bucket.py ..

============== 2 passed, 2 warnings in 75.93s (0:01:15) ==============

```

### CODE REVIEW

File: tests/integration/test_aws/data/test_cases/basic_test_module/cases_inspector_defaults.yaml [hunk 1]
```diff
@@ -2,5 +2,9 @@
   description: Inspector default configurations
   configuration_parameters:
     SERVICE_TYPE: inspector
+    LOG_GROUP_NAME: wazuh-inspector-integration-tests
   metadata:
+    resource_type: log_group
     service_type: inspector
+    log_group_name: wazuh-inspector-integration-tests
```
Comment (fdalmaup) [inline tests/integration/test_aws/data/test_cases/basic_test_module/cases_inspector_defaults.yaml:9]: The log group should not be created for this case, the AWS Inspector service does not make use of it. It must be taken into account in the issues that continue the work of the AWS ITs.

**Review by fdalmaup — CHANGES_REQUESTED** [review]
Respecting the docstrings:
- We should follow the same format for every docstring, for example:
```
  Params
  ------
  - parameters (list): The parameters of the test.
  - metadata (list): The metadata of the test.
```
or 
```
    Parameters
    ----------
    buckets_manager : fixture
        Set of buckets.
    metadata : dict
        Bucket information.
```
Please review the docstrings to follow a specific format.

File: tests/integration/test_aws/data/test_cases/basic_test_module/cases_inspector_defaults.yaml (hunk 1 above)
Comment (EduLeon12) [reply tests/integration/test_aws/data/test_cases/basic_test_module/cases_inspector_defaults.yaml:9]: I agree with this.

File: tests/integration/test_aws/test_custom_bucket.py [hunk 2]
```diff
@@ -223,6 +225,9 @@ def test_custom_bucket_logs(configuration, metadata, load_wazuh_basic_configurat
 
     assert log_monitor.callback_result is not None, ERROR_MESSAGE['failed_start']
 
+    # Give time to the queue to retrieve the messages.
+    time.sleep(30)
```
[the remark this answers is not in the document]
Comment (fdalmaup) [reply tests/integration/test_aws/test_custom_bucket.py:229]: ERRATA:
The problem was due to the incorrect setting of the credentials in the `.aws/credentials` file for the tests. The `default` profile is used when no profile is passed to the AWS module, as explained in the [Boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/1.17.85/guide/credentials.html).

File: tests/integration/test_aws/conftest.py [hunk 3]
```diff
@@ -3,28 +3,34 @@
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 """
-This module contains all necessary components (fixtures, classes, methods)to configure the test for its execution.
+This module contain all necessary components (fixtures, classes, methods) to configure the test for its execution.
```
[the remark this answers is not in the document]
Comment (fdalmaup) [reply tests/integration/test_aws/conftest.py]: To be changed in #22514.

**Pull request merged**